test/gurka/test_match.cc

```diff
@@ -105,4 +107,17 @@ D--3--4--C--5--6--E)";
 
   auto result =
       gurka::match(map, {"2", "6", "3"}, "via", "auto", R"({"penalize_immediate_uturn":false})");
+
+  auto shape =
+      midgard::decode<std::vector<midgard::PointLL>>(result.trip().routes(0).legs(0).shape());
+  // TODO: Remove the duplicate 6 when we fix odin to handle uturn maneuver generation with only one
+  // turn around point
+  auto expected_shape = decltype(shape){
+      map.nodes["2"], map.nodes["B"], map.nodes["C"], map.nodes["6"],
+      map.nodes["6"], map.nodes["C"], map.nodes["3"],
+  };
+  EXPECT_EQ(shape.size(), expected_shape.size());
+  for (int i = 0; i < shape.size(); ++i) {
+    EXPECT_TRUE(shape[i].ApproximatelyEqual(expected_shape[i]));
+  }
```

Comment:
this test case showed the above behavior so we simply expanded it to check the actual shape was what we thought it should be (this is how we found out that the odin bug with uturn maneuvers exists). we decided not to fix the odin uturn bug at this time and instead opted to leave the extra shape point in (at the uturn point) so that we get the correct maneuver